plugins/catalog/src/alpha/pages.tsx

```diff
@@ -89,14 +92,18 @@ export const catalogEntityPage = PageBlueprint.makeWithOverrides({
       loader: async () => {
         const { EntityLayout } = await import('./components/EntityLayout');
 
+        const extraMenuItems = inputs.extraContextMenuItems.map(item =>
+          item.get(coreExtensionData.reactElement),
+        );
+
         type Groups = Record<
           string,
           { title: string; items: Array<(typeof inputs.contents)[0]> }
         >;
 
         const header = inputs.header?.get(
           EntityHeaderBlueprint.dataRefs.element,
-        ) ?? <EntityHeader />;
+        ) ?? <EntityHeader extraMenuItems={extraMenuItems} />;
```

Comment:
then this would probably be more something along the lines of `<EntityHeader contextMenuItems={items} />` where the props of the header has `contextMenuItems?: ContextMenuItem[] | undefined` and where if it's passed in, the default behaviour that it used to have (of rendering a couple of default ones in a fixed set) is disabled, and instead exactly the given items are rendered.

Bonus points for making dividers a possible type of menu item somehow :)